action.yml

```diff
@@ -0,0 +1,104 @@
+name: 'Publish to NeoFS'
+description: 'This action allows you to publish files, tests reports, static web pages and etc. to NeoFS'
+author: 'Neo SPCC'
+branding:
+  icon: 'server'
+  color: 'green'
+inputs:
+  NEOFS_WALLET:
+    description: N3 wallet for NeoFS network access
+    required: true
+  NEOFS_WALLET_PASSWORD:
+    description: N3 wallet password
+    required: true
+  NEOFS_NETWORK_DOMAIN:
+    description: Rpc endpoint domain address
+    required: false
+    default: 'st1.storage.fs.neo.org'
+  NEOFS_HTTP_GATE:
+    description: HTTP Gateway domain address
+    required: false
+    default: 'http.fs.neo.org'
+  STORE_OBJECTS_CID:
+    description: Container ID for your data
+    required: true
+  PATH_TO_FILES_DIR:
+    description: Path to the directory with the files to be pushed
+    required: true
+  NEOFS_ATTRIBUTES:
+    description: User attributes in form of Key1=Value1,Key2=Value2
+    required: false
+  URL_PREFIX:
+    description: Prefix to the url address for each of the files(objects)
+    required: false
+  LIFETIME:
+    description: Number of epochs for object to stay valid
+    required: false
+    default: 0
+
+outputs:
+  OUTPUT_CONTAINER_URL:
+    description: Container URL
+    value: ${{ steps.put_files.outputs.container_url }}
+
+runs:
+  using: composite
+  steps:
+    - name: Download latest stable neofs-cli for uploading reports to NeoFS
+      uses: dsaltares/fetch-gh-release-asset@1.1.1
+      with:
+        repo: 'nspcc-dev/neofs-node'
+        version: 'tags/v0.38.1'
+        file: 'neofs-cli-amd64'
+        target: 'neofs/neofs-cli'
+
+    - name: Chmod latest stable neofs-cli
+      shell: bash
+      run: |
+        sudo chmod a+x neofs-cli
+      working-directory: neofs
+
+    - name: Enable stable neofs-cli
+      shell: bash
+      run: |
+        echo "$(pwd)" >> $GITHUB_PATH
+      working-directory: neofs
+
+    - name: Create wallet
+      shell: bash
+      env:
+        NEOFS_WALLET: ${{ inputs.NEOFS_WALLET }}
+      run: |
+        echo "$NEOFS_WALLET" | base64 -d > wallet.json
+
+    - name: Prepare venv
+      shell: bash
+      id: prepare_venv
+      run: |
+        python3 -m venv venv
+        source venv/bin/activate && pip install -r requirements.txt
+
+    - name: Put files to NeoFS
+      shell: bash
+      id: put_files
+      env:
+        NEOFS_WALLET_PASSWORD: ${{ inputs.NEOFS_WALLET_PASSWORD }}
```

Comment:
It's not a change request, just a comment: I'm not sure it's a good idea to set password as an environment variable. Consider that it's mainnet wallet and mainnet password. Maybe it would be better to pass it directly to the script as an  argument? But still, I'm not sure how safe is it. What do you think?

Reply:
> It's not a change request, just a comment: I'm not sure it's a good idea to set password as an environment variable. Consider that it's mainnet wallet and mainnet password. Maybe it would be better to pass it directly to the script as an argument? But still, I'm not sure how safe is it. What do you think?

I don't see an attack vector here. The secret is still printed as "***" in the github log.
If a user runs something third-party and doesn't monitor the software, it still gives access to the secret.